eng/templates/stages/release.yml

```
@@ -152,14 +152,51 @@ stages:
             draft='--draft'
           fi
 
+          release_json="$(Build.ArtifactStagingDirectory)/release.json"
+
+          echo "{}" \
+            | jq '. += { "release": "$(release)" }'                                \
+            | jq '. += { "channel": "$(releaseChannel)" }'                         \
+            | jq '. += { "tag": "$(releaseTag)" }'                                 \
+            | jq '. += { "sdkVersion": "$(sdkVersion)" }'                          \
+            | jq '. += { "runtimeVersion": "$(runtimeVersion)" }'                  \
+            | jq '. += { "sourceRepository": "https://github.com/dotnet/dotnet" }' \
+            | jq '. += { "sourceVersion": "$(dotnetDotnetCommit)" }'               \
+            > "$release_json"
+
+          echo -e "\n\nGenerated release.json:\n\n"
+          cat "$release_json"
+
+          release_notes="$(Build.ArtifactStagingDirectory)/release_notes.md"
+
+          ./create-announcement-draft.sh                   \
+            --template "source-build-release-notes.md"     \
+            --channel "$(releaseChannel)"                  \
+            $prerelease                                    \
+            --release "$(release)"                         \
+            --release-name "${{ parameters.releaseName }}" \
+            --runtime-version "$(runtimeVersion)"          \
+            --sdk-version "$(sdkVersion)"                  \
+            --tag "$(releaseTag)"                          \
+            --source-version "$(dotnetDotnetCommit)"       \
+            > "$release_notes"
+
+          echo -e "\n\nGenerated release notes:\n\n"
+          cat "$release_notes"
+
+          echo -e "\n\nCreating GitHub release..."
+
           gh release create "$(releaseTag)"         \
+            "$release_json#Release information"     \
```

The label of this file should match the name of the option in the build script.

```suggestion
            "$release_json#Release manifest"     \
```